crates/mcp-client/src/transport/stdio.rs

```diff
@@ -120,6 +120,7 @@ impl StdioTransport {
             .stdout(std::process::Stdio::piped())
             .stderr(std::process::Stdio::inherit())
             .kill_on_drop(true)
+            .process_group(0) // don't inherit signal handling from parent process
```

Comment:
nit: 0 is a reserved value that makes the process group ID equal to the process ID, is that right? (from https://doc.rust-lang.org/std/os/unix/process/trait.CommandExt.html#tymethod.process_group) thats ideal here, but i'd leave another comment line explaining that for future authors

Reply:
good call, added the comment!